File: tasks/implementations/java/Dispatcher.java
public class Dispatcher {
    public static void main(String[] args) throws Exception {
        if ("start".equals(args[0])) {
            String task = args[1];
            String size = args[2];
            int port = Integer.parseInt(System.getenv().getOrDefault("PORT", "8000"));
            switch (task) {
                case "simple_web_server" -> SimpleWebServer.start(size, port, "/fixtures");
                case "rest_api" -> RestApi.start(size, port, "/fixtures");
                case "sqlite_crud" -> SqliteCrud.start(size, port, "/fixtures");
                case "chat_application" -> ChatApplication.start(size, port, "/fixtures");
                case "socket_programming" -> SocketProgramming.start(size, port, "/fixtures");
                case "gui_calculator" -> GuiCalculator.start(size, port, "/fixtures");
                case "data_visualization" -> DataVisualization.start(size, port, "/fixtures");
                case "basic_web_application" -> BasicWebApplication.start(size, port, "/fixtures");
                default -> throw new IllegalArgumentException("unsupported service task: " + task);
            }
            return;
        }
        String task = args[0];
        String size = args[1];
        switch (task) {
            case "sort_integers" -> SortIntegers.run(size, "/fixtures");
            case "matrix_multiplication" -> MatrixMultiplication.run(size, "/fixtures");
            case "csv_parsing" -> CSVParsing.run(size, "/fixtures");
            case "prime_sieve" -> PrimeSieve.run(size, "/fixtures");
            case "file_io_large" -> FileIOLarge.run(size, "/fixtures");
            case "binary_search_tree" -> BinarySearchTree.run(size, "/fixtures");
            case "linear_regression" -> LinearRegression.run(size, "/fixtures");
            case "cli_file_search" -> CliFileSearch.run(size, "/fixtures");
            case "tic_tac_toe" -> TicTacToe.run(size, "/fixtures");
            case "basic_blockchain" -> BasicBlockchain.run(size, "/fixtures");
            case "image_resizing" -> ImageResizing.run(size, "/fixtures");
            case "sentiment_analysis" -> SentimentAnalysis.run(size, "/fixtures");
            case "decision_tree" -> DecisionTree.run(size, "/fixtures");
            case "producer_consumer" -> ProducerConsumer.run(size, "/fixtures");
            case "web_scraper" -> WebScraper.run(size, "/fixtures");
            case "api_client" -> ApiClient.run(size, "/fixtures");
            case "third_party_api" -> ThirdPartyApi.run(size, "/fixtures");
            case "ai_service_integration" -> AiServiceIntegration.run(size, "/fixtures");
            default -> throw new IllegalArgumentException("unsupported task: " + task);
        }
    }
}

File: tasks/implementations/python2/dispatcher.py
import importlib
import os
import sys

DISPATCH = {
    "sort_integers": ("tasks.sort_integers", "run"),
    "matrix_multiplication": ("tasks.matrix_multiplication", "run"),
    "csv_parsing": ("tasks.csv_parsing", "run"),
    "prime_sieve": ("tasks.prime_sieve", "run"),
    "file_io_large": ("tasks.file_io_large", "run"),
    "binary_search_tree": ("tasks.binary_search_tree", "run"),
    "linear_regression": ("tasks.linear_regression", "run"),
    "cli_file_search": ("tasks.cli_file_search", "run"),
    "tic_tac_toe": ("tasks.tic_tac_toe", "run"),
    "basic_blockchain": ("tasks.basic_blockchain", "run"),
    "image_resizing": ("tasks.image_resizing", "run"),
    "sentiment_analysis": ("tasks.sentiment_analysis", "run"),
    "decision_tree": ("tasks.decision_tree", "run"),
    "producer_consumer": ("tasks.producer_consumer", "run"),
    "web_scraper": ("tasks.web_scraper", "run"),
    "api_client": ("tasks.api_client", "run"),
    "third_party_api": ("tasks.third_party_api", "run"),
    "ai_service_integration": ("tasks.ai_service_integration", "run"),
}

SERVICE_DISPATCH = {
    "simple_web_server": ("tasks.simple_web_server", "start"),
    "rest_api": ("tasks.rest_api", "start"),
    "sqlite_crud": ("tasks.sqlite_crud", "start"),
    "chat_application": ("tasks.chat_application", "start"),
    "socket_programming": ("tasks.socket_programming", "start"),
    "gui_calculator": ("tasks.gui_calculator", "start"),
    "data_visualization": ("tasks.data_visualization", "start"),
    "basic_web_application": ("tasks.basic_web_application", "start"),
}


def _resolve(mapping, task_id):
    module_name, attr_name = mapping[task_id]
    module = importlib.import_module(module_name)
    return getattr(module, attr_name)


if __name__ == '__main__':
    if sys.argv[1] == 'start':
        task_id, size = sys.argv[2], sys.argv[3]
        if task_id not in SERVICE_DISPATCH:
            raise SystemExit('unsupported service task: {0}'.format(task_id))
        _resolve(SERVICE_DISPATCH, task_id)(size, int(os.environ.get('PORT', '8000')), '/fixtures')
        raise SystemExit(0)

    task_id, size = sys.argv[1], sys.argv[2]
    if task_id not in DISPATCH:
        raise SystemExit('unsupported task: {0}'.format(task_id))
    _resolve(DISPATCH, task_id)(size, '/fixtures')
